### ISSUE TITLE
[CI] :x-pack:plugin:enrich:test fails

### ISSUE THREAD

**Issue by ywangd** [elastic/elasticsearch#105475]
### CI Link

https://gradle-enterprise.elastic.co/s/us627zmjcye66

### Repro line

N/A

### Does it reproduce?

Didn't try

### Applicable branches

8.12

### Failure excerpt

Look into the console output, I noticed following fatal error
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f5b777fa42e, pid=288874, tid=289337
#
# JRE version: OpenJDK Runtime Environment (21.0.2+13) (build 21.0.2+13-58)
# Java VM: OpenJDK 64-Bit Server VM (21.0.2+13-58, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x5bc42e][thread 289021 also had an error]
  ciReturnTypeEntry::translate_type_data_from(ReturnTypeEntry const*)+0x2e
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/lib/systemd/systemd-coredump %P %u %g %s %t %c %h %e" (or dumping to /opt/local-ssd/buildkite/builds/bk-agent-prod-gcp-1707801153880524692/elastic/elasticsearch-periodic-platform-support/x-pack/plugin/enrich/build/testrun/test/core.288874)
#
# An error report file with more information is saved as:
# /opt/local-ssd/buildkite/builds/bk-agent-prod-gcp-1707801153880524692/elastic/elasticsearch-periodic-platform-support/x-pack/plugin/enrich/build/testrun/test/hs_err_pid288874.log

[error occurred during error reporting (), id 0xb, SIGSEGV (0xb) at pc=0x00007f5b777e7c18]
#
# If you would like to submit a bug report, please visit:
#   https://bugreport.java.com/bugreport/crash.jsp
#
```

The java process seems to encounter an memory issue when compiling the code. Not sure whether it is an infrastructure issue. The error report and reply file can be found in the uploaded artefact. Raising this so that experts in this area can take a look.

**Comment by mark-vieira** [comment]
This is usually indicative of a JDK bug. Here's what the log says:

```
---------------  T H R E A D  ---------------

Current thread (0x00007f5ad024d510):  JavaThread "C1 CompilerThread4" daemon [_thread_in_vm, id=289337, stack(0x00007f5af5dd2000,0x00007f5af5ed2000) (1024K)]


Current CompileTask:
C1:  11475 7556       3       org.elasticsearch.xpack.core.enrich.action.EnrichStatsAction$Response$ExecutingPolicy::equals (63 bytes)
```

That method is pretty anodyne so I'm not sure why the JIT would barf here. Perhaps @ChrisHegarty has some thoughts.

**Comment by thecoop** [comment]
[hs_err_pid288874.log](https://github.com/elastic/elasticsearch/files/14348941/hs_err_pid288874.log)

The full stacktrace is this:

```
Current CompileTask:
C1:  11475 7556       3       org.elasticsearch.xpack.core.enrich.action.EnrichStatsAction$Response$ExecutingPolicy::equals (63 bytes)

Stack: [0x00007f5af5dd2000,0x00007f5af5ed2000],  sp=0x00007f5af5ece170,  free space=1008k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
V  [libjvm.so+0x5bc42e]  ciReturnTypeEntry::translate_type_data_from(ReturnTypeEntry const*)+0x2e
V  [libjvm.so+0x5bcf47]  ciMethodData::load_data()+0x567
V  [libjvm.so+0x5ba7b8]  ciMethod::ensure_method_data()+0x168
V  [libjvm.so+0x4e197f]  GraphBuilder::try_inline_full(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x19f
V  [libjvm.so+0x4dd80d]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x1ad
V  [libjvm.so+0x4e2cd9]  GraphBuilder::try_method_handle_inline(ciMethod*, bool)+0x1a9
V  [libjvm.so+0x4dd7ad]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x14d
V  [libjvm.so+0x4ddcb1]  GraphBuilder::invoke(Bytecodes::Code)+0x451
V  [libjvm.so+0x4df1db]  GraphBuilder::iterate_bytecodes_for_block(int)+0x5cb
V  [libjvm.so+0x4e178d]  GraphBuilder::iterate_all_blocks(bool)+0x6d
V  [libjvm.so+0x4e217e]  GraphBuilder::try_inline_full(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x99e
V  [libjvm.so+0x4dd80d]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x1ad
V  [libjvm.so+0x4e2cd9]  GraphBuilder::try_method_handle_inline(ciMethod*, bool)+0x1a9
V  [libjvm.so+0x4dd7ad]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x14d
V  [libjvm.so+0x4ddcb1]  GraphBuilder::invoke(Bytecodes::Code)+0x451
V  [libjvm.so+0x4df1db]  GraphBuilder::iterate_bytecodes_for_block(int)+0x5cb
V  [libjvm.so+0x4e179d]  GraphBuilder::iterate_all_blocks(bool)+0x7d
V  [libjvm.so+0x4e217e]  GraphBuilder::try_inline_full(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x99e
V  [libjvm.so+0x4dd80d]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x1ad
V  [libjvm.so+0x4e2cd9]  GraphBuilder::try_method_handle_inline(ciMethod*, bool)+0x1a9
V  [libjvm.so+0x4dd7ad]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x14d
V  [libjvm.so+0x4ddcb1]  GraphBuilder::invoke(Bytecodes::Code)+0x451
V  [libjvm.so+0x4df1db]  GraphBuilder::iterate_bytecodes_for_block(int)+0x5cb
V  [libjvm.so+0x4e178d]  GraphBuilder::iterate_all_blocks(bool)+0x6d
V  [libjvm.so+0x4e217e]  GraphBuilder::try_inline_full(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x99e
V  [libjvm.so+0x4dd80d]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x1ad
V  [libjvm.so+0x4e2cd9]  GraphBuilder::try_method_handle_inline(ciMethod*, bool)+0x1a9
V  [libjvm.so+0x4dd7ad]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x14d
V  [libjvm.so+0x4ddcb1]  GraphBuilder::invoke(Bytecodes::Code)+0x451
V  [libjvm.so+0x4df1db]  GraphBuilder::iterate_bytecodes_for_block(int)+0x5cb
V  [libjvm.so+0x4e179d]  GraphBuilder::iterate_all_blocks(bool)+0x7d
V  [libjvm.so+0x4e217e]  GraphBuilder::try_inline_full(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x99e
V  [libjvm.so+0x4dd80d]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x1ad
V  [libjvm.so+0x4e2cd9]  GraphBuilder::try_method_handle_inline(ciMethod*, bool)+0x1a9
V  [libjvm.so+0x4dd7ad]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x14d
V  [libjvm.so+0x4ddcb1]  GraphBuilder::invoke(Bytecodes::Code)+0x451
V  [libjvm.so+0x4df1db]  GraphBuilder::iterate_bytecodes_for_block(int)+0x5cb
V  [libjvm.so+0x4e178d]  GraphBuilder::iterate_all_blocks(bool)+0x6d
V  [libjvm.so+0x4e217e]  GraphBuilder::try_inline_full(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x99e
V  [libjvm.so+0x4dd80d]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x1ad
V  [libjvm.so+0x4e2cd9]  GraphBuilder::try_method_handle_inline(ciMethod*, bool)+0x1a9
V  [libjvm.so+0x4dd7ad]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x14d
V  [libjvm.so+0x4ddcb1]  GraphBuilder::invoke(Bytecodes::Code)+0x451
V  [libjvm.so+0x4df1db]  GraphBuilder::iterate_bytecodes_for_block(int)+0x5cb
V  [libjvm.so+0x4e179d]  GraphBuilder::iterate_all_blocks(bool)+0x7d
V  [libjvm.so+0x4e217e]  GraphBuilder::try_inline_full(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x99e
V  [libjvm.so+0x4dd80d]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x1ad
V  [libjvm.so+0x4ddcb1]  GraphBuilder::invoke(Bytecodes::Code)+0x451
V  [libjvm.so+0x4df1db]  GraphBuilder::iterate_bytecodes_for_block(int)+0x5cb
V  [libjvm.so+0x4e179d]  GraphBuilder::iterate_all_blocks(bool)+0x7d
V  [libjvm.so+0x4e217e]  GraphBuilder::try_inline_full(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x99e
V  [libjvm.so+0x4dd80d]  GraphBuilder::try_inline(ciMethod*, bool, bool, Bytecodes::Code, Instruction*)+0x1ad
V  [libjvm.so+0x4ddcb1]  GraphBuilder::invoke(Bytecodes::Code)+0x451
V  [libjvm.so+0x4df1db]  GraphBuilder::iterate_bytecodes_for_block(int)+0x5cb
V  [libjvm.so+0x4e178d]  GraphBuilder::iterate_all_blocks(bool)+0x6d
V  [libjvm.so+0x4e36e1]  GraphBuilder::GraphBuilder(Compilation*, IRScope*)+0x2d1
V  [libjvm.so+0x4e8dc2]  IRScope::IRScope(Compilation*, IRScope*, int, ciMethod*, int, bool)+0x182
V  [libjvm.so+0x4e8e68]  IR::IR(Compilation*, ciMethod*, int)+0x88
V  [libjvm.so+0x4c31f6]  Compilation::build_hir() [clone .part.0]+0x146
V  [libjvm.so+0x4c4f2c]  Compilation::compile_java_method()+0x3bc
V  [libjvm.so+0x4c50b2]  Compilation::compile_method()+0x112
V  [libjvm.so+0x4c5469]  Compilation::Compilation(AbstractCompiler*, ciEnv*, ciMethod*, int, BufferBlob*, bool, DirectiveSet*)+0x1f9
V  [libjvm.so+0x4c6301]  Compiler::compile_method(ciEnv*, ciMethod*, int, bool, DirectiveSet*)+0xb1
V  [libjvm.so+0x64c847]  CompileBroker::invoke_compiler_on_method(CompileTask*)+0xa97
V  [libjvm.so+0x64f928]  CompileBroker::compiler_thread_loop()+0x698
V  [libjvm.so+0x8f8a78]  JavaThread::thread_main_inner() [clone .part.0]+0xb8
V  [libjvm.so+0xe9e998]  Thread::call_run()+0xa8
V  [libjvm.so+0xcc444a]  thread_native_entry(Thread*)+0xda

siginfo: si_signo: 11 (SIGSEGV), si_code: 1 (SEGV_MAPERR), si_addr: 0x0000000007850fe8
```

**Comment by ChrisHegarty** [comment]
It's definitely a JVM bug - for sure. Will get it reported to OpenJDK.

If we need to, and this becomes problematic, then we can consider whether to exclude the particular method from compilation, e.g. see #103112.   Or if we're feeling really lucky, we can try to "workaround" it, e.g. see https://github.com/apache/lucene/pull/12905

**Comment by ChrisHegarty** [comment]
JDK issue - https://bugs.openjdk.org/browse/JDK-8326356

**Comment by thecoop** [comment]
The JDK issue has been fixed in 21.0.3. If I'm reading it correctly, it's not really to do with the method itself, but a threading issue within the JDK? So excluding that particular method won't fix it, as it could pop up again somewhere else.

**Comment by mark-vieira** [comment]
Sounds like we can close this issue then. There's nothing else to be done here.
